docs/source/usage/trainingmodel.rst

```diff
@@ -5,3 +5,43 @@ Neural networks are powerful machine learning tools in principle capable of
 approximating any function. In MALA, neural networks are built using PyTorch.
 Hyperparameter optimization can be done using optuna and custom routines.
 To see how a model is trained, refer to e.g. ``ex01_singleshot``.
+
+Optimal performance
+*******************
+
+MALA by default uses sub-optimal training parameters to guarantee compatibility
+with a wide range of systems. Users are encouraged to activate advanced
+features for optimal performance. The following is a list of parameters that
+can and should be looked into.
+
+    .. code-block:: python
+
+        parameters = mala.Parameters()
+        """
+        This setting is pretty straightforward - wherever possible, use a GPU.
+        """
+        parameters.use_gpu = False # True: Use GPU
+        """
+        Multiple workers allow for faster data processing, but require
+        additional CPU/RAM power. A good setup is e.g. using 4 CPUs attached
+        to one GPU and setting the num_workers to 4.
+        Please be aware that using multiple workers is currently not supported
+        with horovod.
```

Comment:
Is the limitation for multiple workers and horovod something that is unique to the `FastTensorDataset` introduced or was it already some existing limitation in MALA?

Reply:
This predates the `FastTensorDataset` and is something that @ellisja had told me to look out for. If I recall correctly, the problem is that in order for multiple workers to work correctly, we need more CPUs, but that would lead to horovod launching more ranks? I guess I have not tested it by e.g. requesting 8 CPUs from slurm but then only launching 2 MPI jobs... To be fully honest, I avoid using horovod at the moment because it has given me a lot of problems with running larger jobs, so I think this limitation would not be a problem for me.

Reply:
From a GPU workload perspective, you would typically launch a single process per GPU and then set `num_workers` to a value up to the number of cores you have available per process. For example, if you had a 32 core CPU and 4 GPUs, you might consider launching 4 MPI ranks per node, you could set `num_workers` up to 8 and not oversubscribe CPU resources (though in reality, you'd set it less than 8 to leave room for PyTorch and other threads). 

I wonder if the issue that was brought up in the past had to do with multiple workers using `fork` to create the worker processes and that causing hangs when using Horovod. This is due to an issue in some MPI implementations having issues (e.g. hanging) when a process forks. This can usually be worked around by having PyTorch use `spawn` or `forkserver` to launch processes instead (see https://github.com/horovod/horovod/blob/85619c6851ef4ae3e514e9351bae8a9ca33c2d3b/examples/pytorch/pytorch_imagenet_resnet50.py#L211-L215 as an example)

Reply:
I totally see your point and that sounds reasonable. I think that should already work with MALA if the MPI configuration is correct, I only haven't tried it out.